### PR TITLE
Dump ConnectionManager state on SIGUSR1

### PR DESCRIPTION
**Pull request by coot**

### CODE REVIEW

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs [hunk 1]
```diff
@@ -431,6 +433,7 @@ instance HasSeverityAnnotation (ConnectionManagerTrace addr (ConnectionHandlerTr
       TrConnectionTimeWait {}       -> Debug
       TrConnectionTimeWaitDone {}   -> Debug
       TrConnectionManagerCounters {} -> Info
+      TrState {}                    -> Debug
```
Comment (karknu) [inline cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:436]: This should be at `Info`.

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs [hunk 2]
```diff
@@ -1030,8 +1039,12 @@ instance ToObject (TraceTxSubmissionInbound txid tx) where
 -- TODO: ouroboros-network should provide a newtype wrapper for 'SockAddr'.
 instance ToObject Socket.SockAddr where
     toObject _verb sockAddr =
-      mkObject [ "sockAddr" .= String (pack $ show sockAddr) ]
+      mkObject [ "sockAddr" .= toJSON sockAddr ]
```
Comment (karknu) [inline cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:1042]: I'd prefer if we separated the address from the port. See ac7cf197f .

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs (hunk 1 above)
Comment (coot) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:436]: Yes, I'll fix this.

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs (hunk 2 above)
Comment (coot) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:1042]: ok, I will also need to change the encoding of  `Map SockAddr AbstractState` to a list of triples `(addr, port, state)`.

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs [hunk 3]
```diff
@@ -1177,6 +1190,38 @@ instance ToObject (TracePeerSelection Socket.SockAddr) where
     mkObject [ "kind" .= String "TracePeerSelection"
              , "event" .= show ev ]
 
+instance Aeson.ToJSON AbstractState where
+    toJSON UnknownConnectionSt =
+      Aeson.object [ "kind" .= String "UnknownConnectionSt" ]
+    toJSON ReservedOutboundSt =
+      Aeson.object [ "kind" .= String "ReservedOutboundSt" ]
+    toJSON (UnnegotiatedSt provenance) =
+      Aeson.object [ "kind" .= String "UnnegotatedSt"
```
Comment (bolt12) [inline cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:1199]: Typo: `Unnegotated` -> `UnnegotiatedSt`

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs [hunk 4]
```diff
@@ -1177,6 +1190,38 @@ instance ToObject (TracePeerSelection Socket.SockAddr) where
...
+                   , "dataFlow" .= String (pack . show $ dataFlow)
+                   ]
+    toJSON (InboundSt dataFlow) =
+      Aeson.object [ "kind" .= String "InboundSt"
+                   , "dataFlow" .= String (pack . show $ dataFlow)
+                   ]
+    toJSON OutboundUniSt =
+      Aeson.object [ "kind" .= String "OutboundUniSt" ]
+    toJSON (OutboundDupSt timeoutExpired) =
+      Aeson.object [ "kind" .= String "OutboundUniSt"
```
Comment (bolt12) [inline cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:1213]: Typo, should be `OutboundDupSt` !!

**Review by bolt12 — CHANGES_REQUESTED** [review]
LGTM but the typos are important to know correctly the state!

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs (hunk 2 above)
Comment (coot) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:1042]: Done

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs (hunk 4 above)
Comment (coot) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:1213]: Good catch, a copy paste error.

File: cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs [hunk 5]
```diff
@@ -1342,6 +1410,16 @@ instance (Show addr, Show versionNumber, Show agreedOptions, ToObject addr)
           [ "kind" .= String "ConnectionManagerCounters"
           , "state" .= String (pack . show $ cmCounters)
           ]
+      TrState cmState ->
+        mkObject
+          [ "kind"  .= String "State"
```
Comment (karknu) [inline cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:1415]: Can you change `State` to something more grep:able.

Comment (coot) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs:1415]: Sure, `connectionManagerState` should be easier to grep.

Comment (coot) [reply cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs]: Done

**Pull request merged**